Gems/Atom/RHI/Code/Source/RHI/MultiDeviceStreamingImagePool.cpp

```diff
@@ -197,4 +197,59 @@ namespace AZ::RHI
     {
         return m_descriptor;
     }
+
+    bool MultiDeviceStreamingImagePool::SetMemoryBudget(size_t newBudget)
+    {
+        bool success{true};
+        IterateObjects<StreamingImagePool>([&success, newBudget]([[maybe_unused]] auto deviceIndex, auto deviceStreamingImagePool)
+        {
+            success &= deviceStreamingImagePool->SetMemoryBudget(newBudget);
+        });
+        return success;
+    }
+
+    const HeapMemoryUsage& MultiDeviceStreamingImagePool::GetHeapMemoryUsage(HeapMemoryLevel heapMemoryLevel) const
+    {
+        auto maxUsageIndex{-1};
+        auto maxBudget{0ull};
+        IterateObjects<StreamingImagePool>([&maxUsageIndex, &maxBudget, heapMemoryLevel](auto deviceIndex, auto deviceStreamingImagePool)
+        {
+            const auto& deviceHeapMemoryUsage{deviceStreamingImagePool->GetHeapMemoryUsage(heapMemoryLevel)};
+            if(deviceHeapMemoryUsage.m_budgetInBytes > maxBudget)
+            {
+                maxBudget = deviceHeapMemoryUsage.m_budgetInBytes;
+                maxUsageIndex = deviceIndex;
+            }
+        });
+
+        return GetDeviceStreamingImagePool(maxUsageIndex)->GetHeapMemoryUsage(heapMemoryLevel);
```

Comment:
There are multiple things coalescing here, on the one hand, it is not that easy to add them all up (since we return `HeapMemoryUsage`, which includes multiple members, including `fragmentation` and a string), so it is not immediately obvious what the "sum" of all would look like?
Secondly, we are returning a `const &`, which would then entail additionally caching this struct in the class?
Always returning one per device would also maybe not be the best idea?

Afaik, it is used to set memory budgets in the `RPI/StreamingImagePool` (where it is used in the `SetMemoryBudget` call, so the maximum is not really what we want) and to be displayed in ImGUI (where we likely want to show the per-device usages?).

I'd prefer still returning the maximum here and changing higher-level code if we need something else.

Reply:
Lets update the comment for this method to indicate that it traverses through all the devices and returns the data for the one with biggest peak.